pfcli/configurator/deployment.py

```diff
@@ -0,0 +1,140 @@
+  tag:
+"""
+
+ORCA_CONFIG = """
+# Orca engine config
+orca_config:
+  max_batch_size:
+  max_token_count:
+  kv_cache_size:
+"""
+
+CKPT_CONFIG = """
+# Orca ckeckpoint config
+ckpt_config:
+  version:
+  catagory:
+  data_type:
+"""
+
+SCALER_CONFIG = """
+# Keda scaler config
+scaler_config:
+  min_deployment_count:
+  max_deployment_count:
+"""
+
+
+@dataclass
+class DeploymentInteractiveConfigurator(InteractiveConfigurator[str]):
+    """Deployment template configuration service."""
+
+    ready: bool = False
+    use_specific_image: bool = False
+    use_ckpt_config: bool = False
+
+    def _render(self) -> str:
+        yaml_str = ""
+        if self.use_specific_image:
+            yaml_str += INFERENCE_SERVER_CONFIG
+        if self.use_ckpt_config:
+            yaml_str += CKPT_CONFIG
+
+        return yaml_str
+
+
+@dataclass
+class OrcaDeploymentInteractiveConfigurator(DeploymentInteractiveConfigurator):
+    """Orca deployment template configuration service"""
+
+    use_scaler: bool = False
+
+    def start_interaction(self):
+        self.ready = True
+
+    def render(self) -> str:
+        assert self.ready
+
+        yaml_str = ORCA_CONFIG
+        yaml_str += self._render()
+        if self.use_scaler:
+            yaml_str += SCALER_CONFIG
+
+        return yaml_str
+
+
+def build_deployment_interactive_configurator(
+    engine_type: EngineType,
+) -> DeploymentInteractiveConfigurator:
+    handler_map: Dict[EngineType, Type[DeploymentInteractiveConfigurator]] = {
+        EngineType.ORCA: OrcaDeploymentInteractiveConfigurator,
+    }
+    configurator = handler_map[engine_type]()
+
+    configurator.start_interaction()
+    return configurator
+
+
+class DRCConfigurator(Configurator):
+    """Configurator for default request config."""
+
+    @property
+    def validation_schema(self) -> dict:
+        return {
+            "type": "object",
+            "properties": {
+                "stop": {"type": "array", "items": {"type": "string"}},
+                "stop_tokens": {
+                    "type": "object",
+                    "properties": {
+                        "properties": {
+                            "tokens": {"type": "array", "items": {"type": "integer"}}
+                        },
+                        "required": ["tokens"],
+                    },
+                },
+                "bad_words": {"type": "array", "items": {"type": "string"}},
+                "bad_word_tokens": {
+                    "type": "object",
+                    "properties": {
+                        "properties": {
+                            "tokens": {"type": "array", "items": {"type": "integer"}}
+                        },
+                        "required": ["tokens"],
+                    },
+                },
+            },
+            "allOf": [
+                {"not": {"required": ["stop_tokens", "stop"]}},
+                {"not": {"required": ["bad_words", "bad_word_tokens"]}},
+            ],
```

Comment:
Question) Is this field validating whether either `stop_token` or `stop` are set?

Reply:
This JSON schema says that "stop_tokens" and "stop" are mutually exclusive and "bad_words" and "bad_word_tokens" are also mutually exclusive. Every field is optional but at least one field should exist (`minProperties: 1`).

Reply:
In `tests/configurator/test_deployment.py`, You can see unit tests that check all the possible cases.